### ISSUE TITLE
Provide copy-constructor for value types

### ISSUE THREAD

**Issue by ArneBachmann** [evhub/coconut#325]
Since `data` creates `collections.namedtuples`, we have to make use of `._replace(kw = value)` to create an updated immutable object. This requires knowledge about Python internals, rather than Coconut being direct and helpful. What about:
```
data A(x:int, y:B)
data B(z:float, zz:bool)

a = A(8, B(2., True))

a_old = a._replace(x = 7)  # looks like a private accessor
a_new = A.of(a, x = 7)  # creates a copy of a, but with an updated x
```
or similar something similar like `.copy` or `.update` or even `.replace` on the type/class object.

Maybe a bit Java-ish, but the `._replace` is really ugly.

**Comment by evhub** [comment]
@ArneBachmann Interestingly, I think this and #239 are both trying to solve the same problem, this one from a more OOP perspective, #239 from a more functional perspective. I think it's a problem worth solving, though I'm not sure what syntax I like best for it yet.

Let's say you have
```
data A(x, y)
a = A(1, 2)
```
and you want to convert `a` to `A(1, 3)`. Then, using existing syntax, you could do
```
A(x, y) = a
b = A(x, 3)
```
or, like you mentioned, you could use the `_replace` method to write
```
b = a._replace(y = 3)
```
but with your new syntax, you'd be able to do something like
```
b = a.replace(y = 3)
```
or with the syntax from #239, you'd be able to do something like
```
A(x, y) = a
y = 3
b = A(...)
```

**Comment by evhub** [comment]
Another option worth considering is adding a keyword argument to `makedata` like
```
makedata(A, copy_from=a, y=3)
```

**Comment by ArneBachmann** [comment]
Unless your keyword happens to be `copy_from`!
But thanks for the pointer to the pattern syntax, I'll try it.

**Comment by ArneBachmann** [comment]
BTW, is `...` a syntactic element in `A(...)` referring to all fields of an A object? I don't get it.

**Comment by evhub** [comment]
@ArneBachmann The idea would be for `...` to act similarly to `**locals()`. Also, one thing to note: your comment about an issue if `copy_from` is used as an attribute also applies if `replace` is used as an attribute, since then `.replace` would retrieve that attribute (that's the reason the `_` is there in the first place).

**Comment by ArneBachmann** [comment]
Integration of something like in #349 would close this issue, wouldn't it?
It works just like above mentioned `b = makedata(A, copy_from=a, y=3)` but with any data type and `**kwargs`

**Comment by ArneBachmann** [comment]
Solved by workaround in #349
